cmd/derper/derper.go

```diff
@@ -177,14 +189,42 @@ func main() {
 	s.SetVerifyClientURLFailOpen(*verifyFailOpen)
 	s.SetTCPWriteTimeout(*tcpWriteTimeout)
 
-	if *meshPSKFile != "" {
-		b, err := os.ReadFile(*meshPSKFile)
+	if *dev {
+		k := setec.StaticSecret(os.Getenv("TAILSCALE_DERPER_MESH_KEY"))
```

Comment:
This seems like a weird order to do this check. Let's check the environment variable before we start fiddling with secret values. In this case we don't need a Secret at all:

```go
key, ok := os.LookupEnv("TAILSCALE_DERPER_MESH_KEY")
if !ok || key == "" {
   log.Println("No mesh key specified for dev")
} else if meshKey, err := checkMeshKey(key); err != nil {
   log.Fatalf(...)
} else {
   s.SetMeshKey(meshKey)
   log.Println("Set mesh key from TAILSCALE_DERPER_MESH_KEY")
}
```

or words to that effect.